components/sensor/LPS25H.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 PWrInSpace
+
+#pragma once
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "driver/i2c.h"
+#include "freertos/FreeRTOS.h"
+/*!
+ \brief LPS25H_I2C_ADDR_SA0_L
+      sensor I2C address if SA0 is pulled low
+ */
+#define LPS25H_I2C_ADDR_SA0_L 0b1011100
+/*!
+ \brief LPS25H_I2C_ADDR_SA0_H
+      sensor I2C address if SA0 is pulled high
+*/
+#define LPS25H_I2C_ADDR_SA0_H 0b1011101
+
+/// Pressure reference values
+#define LPS25H_REG_REF_P_XL 0x08
+#define LPS25H_REG_REF_P_L 0x09
+#define LPS25H_REG_REF_P_H 0x0A
+/// Who am I - returns 0xBD
+#define LPS25H_REG_WHO_AM_I 0x0F
+/// Internal average values register
+#define LPS25H_REG_RES_CONF 0x10
+/// Control registers
+#define LPS25H_REG_CTRL_REG1 0x20
+#define LPS25H_REG_CTRL_REG2 0x21
+#define LPS25H_REG_CTRL_REG3 0x22
+#define LPS25H_REG_CTRL_REG4 0x23
+#define LPS25H_REG_INTERRUPT_CFG 0x24
+#define LPS25H_REG_INT_SOURCE 0x25
+/// Status register -
+#define LPS25H_REG_STATUS_REG 0x00
+/// Output pressure registers
+#define LPS25H_REG_PRESS_OUT_XL 0x28
+#define LPS25H_REG_PRESS_OUT_L 0x29
+#define LPS25H_REG_PRESS_OUT_H 0x2A
+/// Temperature output values
+#define LPS25H_REG_TEMP_OUT_L 0x2B
+#define LPS25H_REG_TEMP_OUT_H 0x2C
+/// FIFO functionality
+#define LPS25H_REG_FIFO_CTRL 0x2E
+#define LPS25H_REG_FIFO_STATUS 0x2F
+/// Threshold values for pressure interrupt generation
+#define LPS25H_REG_THS_P_L 0x30
+#define LPS25H_REG_THS_P_H 0x31
+
+#define I2C_MASTER_TIMEOUT_MS 1000
+
+typedef struct {
+  i2c_config_t conf;
+  i2c_port_t port;
+  uint8_t addr;
+} LPS25H;
+
+/*!
```

Comment:
XD po co to

Reply:
A dobra gdybym tego nie przeczytał to bym nie wiedział, że jak nie chcesz żeby to ci stowrzyło i2c to musisz dać zero. Tak dość niestandardowo dawać jako wskaźnik 0, nie wiem czy nie lepiej było by zostać przy NULL, wtedy też nie ma tego rzutowania.

Reply:
ale no i tak po co ta dokumentacja

Reply:
X D